composer/utils/inference.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Inference-related utility functions for model export and optimizations.
+
+Used for exporting models into various formats such ONNX, torchscript etc. and apply optimizations such as fusion.
+"""
+
+import copy
+import logging
+import os
+import tempfile
+from typing import Callable, Optional, Sequence, Union
+
+import torch
+
+from composer.models.base import ComposerModel
+from composer.utils import ObjectStore, dist
+from composer.utils.file_helpers import get_file, is_tar
+from composer.utils.iter_helpers import ensure_tuple
+
+log = logging.getLogger(__name__)
+
+__all__ = [
+    'export_for_inference',
+]
+
+
+def export_for_inference(
+    model: ComposerModel,
+    target_format: str,
+    store_path: str,
+    store_object_store: Optional[ObjectStore] = None,
+    sample_input: Optional[Union[torch.Tensor, Sequence[torch.Tensor]]] = None,
+    surgery_algs: Optional[Union[Callable, Sequence[Callable]]] = None,
+    transforms: Optional[Union[Callable, Sequence[Callable]]] = None,
+    load_path: Optional[str] = None,
+    load_object_store: Optional[ObjectStore] = None,
+) -> None:
+    """Export model for inference from a local file, URI, or cloud object store.
+
+    Args:
+        model (ComposerModel): An instance of ComposerModel. Please note that model is not modified inplace.
+            A copy of the model is made and export related transforms work on this copy.
+        target_format (str):  format to export to. For example, torchscript, onnx, etc.
+        store_path: (str): The path string for storing the exported model.
+        store_object_store (ObjectStore, optional): If the ``path`` is in an object store
+            (i.e. AWS S3 or Google Cloud Storage), an instance of
+            :class:`~.ObjectStore` which will be used
+            to store the exported model. Otherwise, if the store_path is a local filepath, set to ``None``.
```

Comment:
Ambiguous whether "set to None" is an instruction the to the reader or a statement about the internal logic of the function.
```suggestion
            to store the exported model. Set this to ``None`` if ``store_path`` is a local filepath.
```

Reply:
Fixed.